### PR TITLE
CASSANDRASC-35: Add the CONTRIBUTING.md doc with guidelines and best practices

### PR DESCRIPTION
**Pull request by frankgh**
The document delves into the contribution guidelines, source code best practices
and source code style.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+                 .compose(computedChecksum -> {
+                     if (!expectedHash.equals(computedChecksum))
+                         return Future.failedFuture();
+                     return Future.succeededFuture(true);
+                 });
+    }
+  }
+```
+
+A new implementation of `CheksumVerifier` that uses a different hashing algorithm can be injected later.
```
Comment (yifan-c) [inline CONTRIBUTING.md]:
```suggestion
A new implementation of `ChecksumVerifier` that uses a different hashing algorithm can be injected later.
````

**Review by yifan-c — APPROVED** [review]
spot a minor typo

File: CONTRIBUTING.md (hunk 1 above)
Comment (frankgh) [reply CONTRIBUTING.md]: wow! that's a hard one to catch. Thanks

**Comment by frankgh** [comment]
Closed via https://github.com/apache/cassandra-sidecar/commit/bc219cbf75bdbfdc7a95b3160ef17332c9274b44

**Pull request closed without merging**